py/server/tests/test_exec_ctx.py

```diff
@@ -0,0 +1,67 @@
+#
+#     Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+import concurrent.futures
+import threading
+import unittest
+
+from deephaven import DHError
+from deephaven import empty_table
+from deephaven.exec_ctx import ExecutionContext
+
+
+class ExecCtxTestCase(unittest.TestCase):
+    def test_exec_ctx_threads(self):
+
+        thread_results = {}
+
+        main_exec_ctx = ExecutionContext.get_exec_ctx()
+
+        def thread_update(p: int):
+            t = empty_table(1)
+            with main_exec_ctx:
+                t2 = t.update("X = p*p")
+            thread_results[p] = t2.to_string()
+
+            with self.assertRaises(DHError):
+                t3 = t.update("Y = p * p * p")
+
+            ExecutionContext.set_exec_ctx(main_exec_ctx)
+            t3 = t.update("Y = p * p * p")
+
+        threads = []
+        for i in range(4):
+            threads.append(threading.Thread(target=thread_update, args=(i,)))
+
+        for th in threads:
+            th.start()
+            th.join()
+
+        self.assertTrue(thread_results)
+        for i, rlt in thread_results.items():
+            self.assertIn(str(i * i), rlt)
+
+    def test_thread_pool_executor(self):
+        thread_results = {}
+
+        main_exec_ctx = ExecutionContext.get_exec_ctx()
+
+        def thread_update(p: int):
+            t = empty_table(1)
+            with main_exec_ctx:
```

Comment:
Throughout the tests, there is a pattern that looks like:
```
main_exec_ctx = ExecutionContext.get_exec_ctx()
...
def thread_func(...):
    ...
    with main_exec_ctx:
        query stuff
    ....
```
or
```
main_exec_ctx = ExecutionContext.get_exec_ctx()
...
def thread_func(...):
    ExecutionContext.set_exec_ctx(main_exec_ctx)
    ...
```
Is there a clean way to condense the same logic into a decorator, so that the user has the option to just do one of the following?
```
@exec_context
def thread_func(...):
    ...
```
or 
```
@exec_context(specific_context)
def thread_func(...):
    ...
```
It may be tricky to do this because the thread func is probably called on some other thread, making the context difficult to obtain from the point the function is called.  I suspect that this will result in many use cases that look like the following since the context needs to be handed around.  I guess it will end up looking like Go code where `context.Context` is everywhere in user code.
```
def thread_func(exec_ctx, ...):
    ...
threading.Thread(target=thread_func, args=(exec_ctx,...,))
```